S3 boto upload

rcrowder opened the pull request:
Fixes #570 

Changes to use `deploy-wheel-to-s3.py` script in AV and Travis.

rcrowder commented:
@rhyolight Can you check the S3 SSL certificate and see if it has expired?

rcrowder commented:
@rhyolight Looks like a deeper issue. See https://github.com/boto/boto/issues/2836

rcrowder commented:
PS: The following gets further (locally)..

conn = boto.connect_s3(is_secure=True,calling_format=boto.s3.connection.OrdinaryCallingFormat())
conn.host='s3-us-west-2.amazonaws.com'
artifactsBucket = conn.get_bucket('artifacts.numenta.org')

But ends with a ```socket.gaierror: [Errno 11004] getaddrinfo failed```

rhyolight commented:
Hm... I'm pretty sure the cert has not expired because we use this S3
bucket for artifacts from linux builds, too.

---------
Matt Taylor
OS Community Flag-Bearer
Numenta

On Tue, Nov 17, 2015 at 9:58 AM, Richard Crowder <notifications@github.com>
wrote:

> PS: The following gets further (locally)..
>
> conn =
> boto.connect_s3(is_secure=True,calling_format=boto.s3.connection.OrdinaryCallingFormat())
> conn.host='s3-us-west-2.amazonaws.com'
> artifactsBucket = conn.get_bucket('artifacts.numenta.org')
>
> But ends with a socket.gaierror: [Errno 11004] getaddrinfo failed
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/numenta/nupic.core/pull/692#issuecomment-157453279>.
>

rcrowder commented:
Yes, the errors from S3 / boto can lead in false directions. It's a deeper DNS, regions, or access control  issue. Looking into at the moment, along with other things..

rcrowder commented:
@scottpurdy cc @rhyolight This change started off as a way to get the boto based python script shared  between AppVeyor and Travis. Unfortunately the decoding of secure environment variables has hampered this on AV (https://ci.appveyor.com/tools/encrypt shows where decryption takes place). So although I had partial success running boto, it looks tricky/impossible to pass in secure variables for the  AWS secret key. As such I've reverted back to standard S3 deployment. And am hoping these changes   will get deployment working again.
requirements.txt change for numpy is to cope with the version we currently use/need on Windows. I'll check why 1.9.2 has been forced..

I moved the boto script to the /ci directory. Should this be moved back to /ci/travis ? If not, then this is ready for review.

scottpurdy commented:
> I'll check why 1.9.2 has been forced..

What does this mean?

But I'll merge now.

rcrowder commented:
Access to a 1.9.2 version of NumPy was not easily available with MinGWPy. But carlkl pointed us today to somewhere get a pre-built 1.9.2 version, see convo here https://github.com/numenta/nupic.core/issues/685
Issues with NumPy look mainly related to pip; https://github.com/numenta/nupic.core/search?q=1.9.2&type=Issues

rcrowder commented:
@scottpurdy Excellent! Thanks. The resulting wheel from this on S3 now installs nicely on Windows. Although having `>=` it now tries to pull in NumPy 1.10.1 from PyPi Placing more reliance on whether a user has numpy installed before pip installing nupic.bindings :( Another hurdle to vault..

scottpurdy commented:
Oh I didn't realize that it switch to `>=`. Can we change it to `==`?

rcrowder commented:
@scottpurdy Sorry about that :( It 'was' an issue supporting NumPy 1.9.2 and above. But with carlkl's advice on another GH issue here, looks like that concern is no longer. It can be reverted back to ```==1.9.2```. Let me know if that is something for me to throw in via a new PR, or something you can do quicker?

scottpurdy commented:
@rcrowder - thx, changing back in #700
